cmd/ingester/app/flags.go

```diff
@@ -63,14 +65,17 @@ const (
 	DefaultEncoding = kafka.EncodingProto
 	// DefaultDeadlockInterval is the default deadlock interval
 	DefaultDeadlockInterval = 1 * time.Minute
+	// DefaultMaxOutOfOrderOffsets is the default max out of ourder offsets to maintain before restarting the processor
+	DefaultMaxOutOfOrderOffsets = 500000
```

Comment:
How do we determine this default value?

Reply:
I used an average message rate of 10k qps, so that this list would be exhausted in 50s. 
These numbers are arbitrary - using a number that is too small would cause churn by closing and recreating partition consumers unnecessarily

Reply:
having some explanation of the default value in the comment would be good.

Reply:
Perhaps it's something we can tweak at runtime?